crates/astria-sequencer/src/accounts/component.rs

```diff
@@ -29,14 +29,16 @@ impl Component for AccountsComponent {
     where
         S: accounts::StateWriteExt + assets::StateReadExt,
     {
-        let native_asset = state
-            .get_native_asset()
-            .await
-            .wrap_err("failed to read native asset from state")?;
-        for account in app_state.accounts() {
-            state
-                .put_account_balance(&account.address, &native_asset, account.balance)
-                .wrap_err("failed writing account balance to state")?;
+        if !app_state.accounts().is_empty() {
```

Comment:
This isn't really needed is it?  I'd be inclined to remove it for clarity, but not a big deal.

Reply:
I need the conditional because I only need the native asset (fetched on the line below) if the accounts are set. If there are no accounts in genesis (and no native asset), then I don't want to potentially fail init chain on that call.

Is that what you meant or am I overlooking something?

Reply:
No - you're correct.  We do need the `if` block.